core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepViewer.java

```diff
@@ -0,0 +1,27 @@
+package ca.jrvs.apps.grep;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+public class JavaGrepViewer {
+
+  private final static Logger LOGGER = Logger.getLogger(JavaGrepViewer.class.getName());
+
+    public static void main(String[] args) {
+    JavaGrepLambdaImp javaGrep = new JavaGrepLambdaImp();
+    if (args.length != 3) {
+      LOGGER.info("Insufficient Arguments");
+    } else {
+      javaGrep.setRegex(args[0]);
+      javaGrep.setRootPath(args[1]);
+      javaGrep.setOutFile(args[2]);
+    }
+
+    try {
+      javaGrep.process();
+    } catch (IOException e) {
+      LOGGER.info("Unable to read");
+    }
```

Comment:
logger.info(message, e)
make sure you pass the caught exception so stack track is tracked